todo/todo.py

```diff
@@ -183,344 +173,370 @@ async def showsettings(self, ctx):
+                        else:
+                            removed += 1
+            msg = "Done."
+            detailed = await self.config.user(ctx.author).detailed_pop()
+            if detailed:
+                if removed:
+                    msg += f"\nRemoved: {removed} {'todos' if removed > 1 else 'todo'}"
+                if failed:
+                    msg += f"\nFailed to remove: {failed} {'todos' if failed > 1 else 'todo'}"
+            await ctx.send(msg)
+            await self._maybe_autosort(ctx)
+
+    @todo.command(aliases=["delall"])
+    async def removeall(self, ctx):
+        """Remove all of your todos"""
+        await ctx.send(
+            "WARNING, this will remove **ALL** of your todos. Would you like to remove your todos? (y/n)"
+        )
+        pred = MessagePredicate.yes_or_no(ctx)
+        try:
+            await self.bot.wait_for("message", check=pred)
+        except asyncio.TimeoutError:
+            await ctx.send("Okay, I won't delete your todos")
+        else:
+            if pred.result is False:
+                await ctx.send("Okay, I won't delete your todos")
+            else:
+                await self.config.user(ctx.author).todos.clear()
+                await ctx.send("Removed your todos.")
 
     @todo.command(name="list")
     async def todo_list(self, ctx):
-        """List your todo reminders"""
+        """List your current todos!"""
         todos = await self.config.user(ctx.author).todos()
-        md = await self.config.user(ctx.author).use_md()
-        if len(todos):
-            todos = await self.number(item=todos)
-            if await self.config.user(ctx.author).combine_lists():
-                if not (comp := await self.config.user(ctx.author).completed()):
-                    pass
-                else:
-                    if not md:
-                        comp = await self.cross_lists(items=comp)
-                    completed = await self.number(comp)
-                    completed.insert(0, "❎ Completed todos")
-                    todos.extend(completed)
-            await self.page_logic(ctx, todos)
+        if not todos:
+            await ctx.send(_no_todo_message.format(prefix=ctx.clean_prefix))
         else:
-            await ctx.send(
-                (
-                    f"You don't have any todos!"
-                    f"\nYou can add one using "
-                    f"`{ctx.clean_prefix}todo add <todo>`"
-                )
-            )
-
-    @todo.command()
-    async def sort(self, ctx: commands.Context):
-        """Sort your todos alphabetically"""
-        if not len(await self.config.user(ctx.author).todos()):
-            return await ctx.send("You don't have any todos!")
-        msg = await ctx.send("Would you like to sort your todos in reverse?")
+            todos = await self._number_lists(todos)
+            if await self.config.user(ctx.author).combined_lists():
+                if (c := await self.config.user(ctx.author).completed()) :
```

Comment:
Not familiar with the walrus operator, eitherway there's no need for this space here, unless its for the walrus operator (which I doubt).

```suggestion
                if (c := await self.config.user(ctx.author).completed()):
```